packages/core/src/command/consolidated/search/fhir-resource/ingest-lexical.ts

```diff
@@ -100,7 +101,17 @@ async function getConsolidatedBundle({
 }): Promise<Bundle<Resource>> {
   const { log } = out(`getConsolidatedBundle - cx ${cxId}, pt ${patientId}`);
 
-  const consolidated = await getConsolidatedFile({ cxId, patientId });
+  const consolidated = await executeWithRetries(() => getConsolidatedFile({ cxId, patientId }), {
+    shouldRetry: (res, error: unknown) => {
+      if (!res) return true;
+      if (!error) return false;
+      if (!isRetriableError(error)) return false;
+      return true;
+    },
+    initialDelay: 1_000,
+    maxAttempts: 3,
+    log,
+  });
```

Comment:
Almost a rewrite of the underlying S3Utils code called by `getConsolidatedFile` so we can retry on 404/NotFound - https://metriport.slack.com/archives/C04T256DQPQ/p1749433999276159?thread_ts=1749096627.350379&cid=C04T256DQPQ